Carbon/Functions/Install-FileShare.ps1

```diff
@@ -175,8 +175,17 @@ function Install-CFileShare
             New-Item -Path $Path -ItemType Directory -Force | Out-String | Write-Verbose
         }
     
+        $shareClass = Get-CCimClass -Class 'Win32_Share'
         Write-Verbose -Message ('[SHARE] [{0}]              Sharing {1}' -f $Name,$Path)
-        $result = ([wmiclass]"root\cimv2:Win32_Share").Create( $Path, $Name, 0, $null, $Description, $null, $shareSecurityDescriptor )
+
+        if( Test-CCimAvailable )
+        {
+            $result = ([wmiclass]"root\cimv2:Win32_Share").Create( $Path, $Name, 0, $null, $Description, $null, $shareSecurityDescriptor )

```

Comment:
I think you can use `Get-CCimClass -Name 'Win32_Share`' instead of `[wmiclass]"root\cimv2:Win32_Share"`.

Reply:
I believe this needs to be kept as is...

I spent a good few hours last night trying to get this to work as it is intended. When using CIM you need to call a method on a CIM class by calling Invoke-CImMethod like so:

 $result = Invoke-CimMethod -CimClass $shareClass -MethodName 'Create' -Arguments @{Path = $Path; Name = 
 Name; Type = 0; Description = $Description; Access = $shareSecurityDescriptor}

However this gives a type mismatch on the Access parameter. This parameter accepts a Win32_SecurityDescriptor so I revised some code to make sure $shareSecurityDescriptor  is an instance of Win32__SecurityDescriptor but still doesn't work as intended. I have this code set aside if you want to see it.